### PR TITLE
Replace react native datepicker with community version for RN 0.62

### PR DESCRIPTION
**Pull request by 0608jennylee**
used this one --> https://github.com/xgfe/react-native-datepicker/pull/436/files

another one that tries to do the upgrade --> https://github.com/xgfe/react-native-datepicker/pull/429/files

Migration to @react-native-community/datetimepicker: https://github.com/react-native-datetimepicker/datetimepicker#migration-from-the-older-components

### CODE REVIEW

File: datepicker.js [hunk 1]
```diff
@@ -262,37 +238,13 @@ class DatePicker extends Component {
     if (Platform.OS === 'ios') {
       this.setModalVisible(true);
     } else {
-
-      const {mode, androidMode, format = FORMATS[mode], minDate, maxDate, is24Hour = !format.match(/h|a/)} = this.props;
-
+      const {mode} = this.props;
       // 选日期
```
Comment (dekpient) [inline datepicker.js]: what does it meannnn Jenny?!?!

Comment (0608jennylee) [reply datepicker.js]: its means choose date

Comment (dekpient) [reply datepicker.js]: you have to translate AllTheThingsss 😛

Comment (0608jennylee) [reply datepicker.js]: noooooooooo

File: datepicker.js [hunk 2]
```diff
@@ -262,37 +237,13 @@ class DatePicker extends Component {
...
-        // 选日期和时间
-
-        DatePickerAndroid.open({
-          date: this.state.date,
-          minDate: minDate && this.getDate(minDate),
-          maxDate: maxDate && this.getDate(maxDate),
-          mode: androidMode
-        }).then(this.onDatetimePicked);
+      const {mode} = this.props;
+      // Choose date
```
Comment (cml35) [inline datepicker.js:241]: Is this comment still needed?

File: datepicker.js [hunk 3]
```diff
@@ -6,22 +6,26 @@ import {
...
   'date': 'YYYY-MM-DD',
   'datetime': 'YYYY-MM-DD HH:mm',
   'time': 'HH:mm'
 };
 
+const DATE_MODE = 'date';
+const TIME_MODE = 'time';
+const DATETIME_MODE = 'datetime';
+const isAndroid = Platform.OS === 'android';
+
```
Comment (cml35) [inline datepicker.js:28]: Do we need an isIOS equivalent?

File: datepicker.js [hunk 4]
```diff
@@ -197,53 +199,26 @@ class DatePicker extends Component {
...
-    if (action !== DatePickerAndroid.dismissedAction) {
-      this.setState({
-        date: new Date(year, month, day)
-      });
-      this.datePicked();
-    } else {
+  onDatetimePicked(event, date) {
+    const {mode} = this.props;
+    const {androidDatePickerMode} = this.state;
+    // cancel pressed
```
Comment (cml35) [inline datepicker.js:205]: Is this comment still needed?

File: datepicker.js [hunk 5]
```diff
@@ -197,53 +199,26 @@ class DatePicker extends Component {
...
-        hour: timeMoment.hour(),
-        minute: timeMoment.minutes(),
-        is24Hour: is24Hour,
-        mode: androidMode
-      }).then(this.onDatetimeTimePicked.bind(this, year, month, day));
-    } else {
-      this.onPressCancel();
-    }
-  }
+    // confirm pressed
```
Comment (cml35) [inline datepicker.js:212]: Likewise here, is the comment still needed?

File: datepicker.js [hunk 6]
```diff
@@ -197,53 +199,26 @@ class DatePicker extends Component {
...
+    const timePickerVisible = androidDatePickerMode === DATE_MODE && mode === DATETIME_MODE;
+    const _androidDatePickerMode = timePickerVisible ? TIME_MODE : undefined;
```
Comment (cml35) [inline datepicker.js:214]: Is it safe to have an undefined state i.e. for _androidDatePickerMode?

File: datepicker.js [hunk 7]
```diff
@@ -262,37 +237,13 @@ class DatePicker extends Component {
...
-          minDate: minDate && this.getDate(minDate),
-          maxDate: maxDate && this.getDate(maxDate),
-          mode: androidMode
-        }).then(this.onDatetimePicked);
+      const {mode} = this.props;
+      // Choose date
+      if (mode === DATE_MODE || mode === DATETIME_MODE) {
+        this.setState({androidDatePickerMode: DATE_MODE});
+      } else if (mode === TIME_MODE) {
+        // Choose time
```
Comment (cml35) [inline datepicker.js:245]: Comment needed?

File: datepicker.js [hunk 8]
```diff
@@ -6,22 +6,26 @@ import {
...
 
 const FORMATS = {
   'date': 'YYYY-MM-DD',
   'datetime': 'YYYY-MM-DD HH:mm',
   'time': 'HH:mm'
 };
 
+const DATE_MODE = 'date';
+const TIME_MODE = 'time';
+const DATETIME_MODE = 'datetime';
```
Comment (aryo) [inline datepicker.js:26]: make enum

File: datepicker.js [hunk 9]
```diff
@@ -433,6 +386,16 @@ class DatePicker extends Component {
               </TouchableComponent>
             </View>
           </Modal>}
+         {(isAndroid && !!androidDatePickerMode)
```
Comment (aryo) [inline datepicker.js:389]: have a separate state for visibility

File: datepicker.js (hunk 6 above)
Comment (0608jennylee) [reply datepicker.js:214]: removing the undefined adding a state for androidDatePickerVisible

File: datepicker.js [hunk 10]
```diff
@@ -6,22 +6,26 @@ import {
...
-  DatePickerAndroid,
-  TimePickerAndroid,
   DatePickerIOS,
   Platform,
   Animated,
   Keyboard
 } from 'react-native';
 import Style from './style';
 import Moment from 'moment';
+import DateTimePicker from '@react-native-community/datetimepicker';
```
Comment (nhig564) [inline datepicker.js:16]:
```suggestion
} from 'react-native';
import Moment from 'moment';
import DateTimePicker from '@react-native-community/datetimepicker';

import Style from './style';
```

sorry just bugged me 🤣

**Review by nhig564 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**